src/audio/module_adapter/module/volume/volume.c

```diff
@@ -551,6 +551,12 @@ static int volume_init(struct processing_module *mod)
 	uint8_t channel;
 	uint32_t instance_id;
 
+	channels_count = mod->priv.cfg.base_cfg.audio_fmt.channels_count;
+	if (channels_count > SOF_IPC_MAX_CHANNELS) {
```

Comment:
does this have to be checked twice? Can the configuration be supplied both in `.init()` and in `.set_configuration()`?

Reply:
@lyakh Given they are separate ops, it seems hard to be absolutely sure and seems safer to check.